### ISSUE TITLE
Unclear Migration Path for enums after move from 3.0.0-dev..8 -> .12

### ISSUE THREAD

**Issue by alevinetx** [isar/isar#595]
I had existing models with enums and TypeConverters.  Now it looks like TypeConverters have been removed, but I'm not seeing the migration path.
I removed the converter annotation, and now I'm getting  ```IsarError: SchemaError: Could not deserialize schema JSON``` errors.  I read through #583 and tried to dig through the source, but I admit I'm completely lost how to proceed.

Thank you !

class:
```
@enumerated 
//@EnumConverter  : this is no longer a valid annotation as TypeConverter has been removed
EnumeratedType someField = EnumeratedType.A;
```

enum:
```
enum EnumeratedType {
A,
B,
C
}
```

**Comment by leisim** [comment]
This has been fixed in the newest version 🙂

**Comment by alevinetx** [comment]
I've installed and regenerated with .13 and still getting the same problem on app start.

I rolled back to 8 and have no problems.   

Is there something I can do to (easily) diagnose/debug what's going on?

Thank you!

```
flutter: ╔══════════════════════════════════════════════╗
flutter: ║             ISAR CONNECT STARTED             ║
flutter: ╟──────────────────────────────────────────────╢
flutter: ║ Open the link in Chrome to connect to the    ║
flutter: ║ Isar Inspector while this build is running.  ║
flutter: ╟──────────────────────────────────────────────╢
flutter: ║ https://inspect.isar.dev/3.0.0-dev.13/#/..... ║
flutter: ╚══════════════════════════════════════════════╝
[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: IsarError: SchemaError: Could not deserialize schema JSON
````

**Comment by leisim** [comment]
Could you please share me your collection class? And even more helpful would be a database file that fails to migrate.

**Comment by alevinetx** [comment]
There's nothing special about these.  A simple 1->many (parent->child) relation with an enum in the (many)child.


parent_model.dart
```
import 'package:isar/isar.dart';
import 'child_model.dart';

part 'parent_model.g.dart';

@Collection()
class ParentModel {
  ParentModel({ required this.startDate});
  Id? id;
  DateTime startDate;
  bool locked = false;

  final parentals = IsarLinks<ChildModel>();

```

child_model.dart
```
import 'package:isar/isar.dart';
import 'parent_model.dart';

part 'child_model.g.dart';

@Collection()
class ChildModel {
  Id? id;
  int itemIndex = 0;

  @enumerated
  ChildType type = ChildType.A;
  String notes  = '';

  @Backlink(to: 'parentals')
  final parental = IsarLink<ParentModel>();
}

enum ChildType {
  A,
  B,
  C
}
```

**Comment by alevinetx** [comment]
Created a new project to see if my problem with adding a linked object and not having it visible was resolved in > .8;  the output from ```printing children``` is not what I would expect.


And I recreated the problem above, and it doesn't appear to be specific to annotations, as I've removed them entirely.  Notice I've never save any records.

I've attached the 3 source files, and 2 zips representing the DB files after program termination. I did not delete the DB files when rebuilding from 8->13.

I hope this helps!

1:  Built and ran with .8
```
flutter: created p1
flutter: created c1
flutter: added c1 to p1, true
flutter: printing children
flutter: done printing children
```

2:  Built and ran with .13


```
[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: IsarError: SchemaError: Could not deserialize schema JSON
```

main.dart
```
void main() async {

  await initcode();

  runApp(const MyApp());
}
....
```

pubspec
```
dependencies:
  flutter:
    sdk: flutter

  isar: 3.0.0-dev.13
  isar_flutter_libs: 3.0.0-dev.13

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^2.0.0

  isar_generator: 3.0.0-dev.13
  build_runner: any
```

[lib.zip](https://github.com/isar/isar/files/9446806/lib.zip)

[isartest-8.zip](https://github.com/isar/isar/files/9446798/isartest-8.zip)
[isartest-13.zip](https://github.com/isar/isar/files/9446799/isartest-13.zip)

**Comment by alevinetx** [comment]
Q:   Is the intended behavior to not see a child in the parent's set until after records have been saved?  This is the issue I was wondering if > .8 had solved, and thus led me to this issue.

1. new parent
2. new child
3. add child to parent links
4. parent.links.count == 1
5. iterate through parent.links
6. --> 0 elements ??  I expect 1
7. save parent +links
8. iterate through parent.links
9. --> 1 element

Thank you for the great work!

I have other questions, like a delete+links cascade, but one thing at a time. 🏳️‍🌈

**Comment by leisim** [comment]
This has been fixed in `3.0.0-dev.14`
